File: src/code-samples/langsmith/smithdb-migration/runs-add-to-queue-before.ts
// :snippet-start: runs-add-to-queue-before-js
// :codegroup-tab: Before
import { Client } from "langsmith";

const client = new Client();
let queueId = "<queue-id>";
const runs = [];
for await (const run of client.listRuns({ projectName: "default", limit: 5 })) {
  runs.push(run);
}
// :remove-start:
const queue = await client.createAnnotationQueue({
  name: `docs-smithdb-migration-${crypto.randomUUID()}`,
});
queueId = queue.id;
if (runs.length === 0) {
  throw new Error("expected at least one run in the 'default' project");
}
// :remove-end:
await client.addRunsToAnnotationQueue(
  queueId,
  runs.map((run) => run.id),
);
// :snippet-end:

// :remove-start:
console.log("✓ runs-add-to-queue-before validated");
// :remove-end:

File: src/code-samples/langsmith/smithdb-migration/runs-add-to-queue-before.go
// :snippet-start: runs-add-to-queue-before-go
// :codegroup-tab: Before
package main

import (
	"context"
	"time"

	"github.com/langchain-ai/langsmith-go"
)

// :remove-start:
func main() {
// :remove-end:
ctx := context.Background()
client := langsmith.NewClient()

queueID := "<queue-id>"
projectID := "<project-id>"
// :remove-start:
sessions, err := client.Sessions.List(ctx, langsmith.SessionListParams{
	Name:  langsmith.F("default"),
	Limit: langsmith.F(int64(1)),
})
if err != nil {
	panic(err.Error())
}
projectID = sessions.Items[0].ID
queue, err := client.AnnotationQueues.AnnotationQueues(ctx, langsmith.AnnotationQueueAnnotationQueuesParams{
	Name: langsmith.F("docs-smithdb-migration-" + time.Now().Format("20060102150405.000000000")),
})
if err != nil {
	panic(err.Error())
}
queueID = queue.ID
// :remove-end:
found, err := client.Runs.Query(ctx, langsmith.RunQueryParams{
	Session: langsmith.F([]string{projectID}),
	Limit:   langsmith.F(int64(5)),
})
// :remove-start:
if err != nil {
	panic(err.Error())
}
// :remove-end:
runIDs := make([]string, len(found.Runs))
for i, run := range found.Runs {
	runIDs[i] = run.ID
}
_, err = client.AnnotationQueues.Runs.New(ctx, queueID, langsmith.AnnotationQueueRunNewParams{
	Body: langsmith.AnnotationQueueRunNewParamsBodyRunsUuidArray(runIDs),
})
// :remove-start:
if err != nil {
	panic(err.Error())
}
}

// :remove-end:
// :snippet-end:
